### ISSUE TITLE
TypeScript error: Type 'undefined' is not assignable to type 'TeamColor'

### ISSUE THREAD

**Issue by salcode** [samdemaeyer/codenames-pictures#6]
When running the latest code on `master` (commit 8b351eb), I'm getting the following error when I run `yarn start`

```
codenames-pictures/src/routes/Game.tsx
TypeScript error in codenames-pictures/src/routes/Game.tsx(100,7):
Type '"red" | "blue" | "green" | null | undefined' is not assignable to type 'TeamColor'.
  Type 'undefined' is not assignable to type 'TeamColor'.  TS2322

     98 |       cards,
     99 |       teams,
  > 100 |       startingTeam,
        |       ^
    101 |       setStartingTeam,
    102 |       score,
    103 |       newGame,
```

Based on my suspicion that a new version of TypeScript might be the culprit, I found modifying the typescript dependency in [package.json](https://github.com/samdemaeyer/codenames-pictures/blob/8b351eb485ec9d3973df807250cde8b2025987c8/package.json#L23) from

```
"typescript": "^3.8.3"
```

to 

```
"typescript": "~3.8.3"
```

resolves the issue (without modifying any of your source code).

## Thank You

Along with this, I wanted to thank you for this project. For months now my family has been holding weekly virtual game nights. Initially, we were using the physical copy of Codenames Pictures that I own. I would take pictures and share them in our Zoom call. This project has made it much easier to play, so from all my family members - Thank you!

**Comment by samdemaeyer** [comment]
Hi @salcode .. there are indeed some errors related to typescript. I have upgraded all dependencies and attempted to fix the typing errors. As @maja-borzemska is the typescript expert 😏 I will check with her to review.

https://github.com/samdemaeyer/codenames-pictures/pull/9

Thank for the compliments around the game, we have been playing the game with colleagues before the COVID times, then out of pure boredness we have decided to build it. It's nice to know other people are using the same as well. 

Anyway, hopefully, the fix will be merged soon.
